scripts/styleChecker.fsx

```diff
@@ -0,0 +1,263 @@
+            Command = "git"
+            Arguments = "restore ."
+        },
+        Echo.Off
+    )
+    |> ignore
+
+let InstallPrettier(version: string) =
+    Process.Execute(
+        {
+            Command = "npm"
+            Arguments = $"install prettier@{version}"
+        },
+        Echo.Off
+    )
+    |> ignore
+
+let PrintProcessResult (processResult: ProcessResult) (suggestion: string) =
+    let errMsg =
+        sprintf
+            "Error when running '%s %s'"
+            processResult.Details.Command
+            processResult.Details.Args
+
+    match processResult.Result with
+    | Success output -> Console.WriteLine output
+    | Error(_, output) ->
+        if processResult.Details.Echo = Echo.Off then
+            output.PrintToConsole()
+            Console.WriteLine()
+            Console.Out.Flush()
+
+        let fullErrMsg = errMsg + Environment.NewLine + suggestion
+        Console.Error.WriteLine fullErrMsg
+
+    | WarningsOrAmbiguous output ->
+        if processResult.Details.Echo = Echo.Off then
+            output.PrintToConsole()
+            Console.WriteLine()
+            Console.Out.Flush()
+
+        let fullErrMsg = sprintf "%s (with warnings?)" errMsg
+        Console.Error.WriteLine fullErrMsg
+
+let GetProcessExitCode(processResult: ProcessResult) : int =
+    match processResult.Result with
+    | Success output -> 0
+    | _ -> 1
+
+let CheckStyleOfFSharpFiles(rootDir: DirectoryInfo) : int =
+    let suggestion =
+        "Please style your F# code using: `dotnet fantomless --recurse .`"
+
+    GitRestore()
+
+    let exitCode =
+        if ContainsFiles rootDir "*.fs" || ContainsFiles rootDir ".fsx" then
+            StyleFSharpFiles()
+            let processResult = GitDiff()
+            PrintProcessResult processResult suggestion
+            GetProcessExitCode processResult
+
+        else
+            0
+
+    exitCode
+
+let CheckStyleOfTypeScriptFiles(rootDir: DirectoryInfo) : int =
+    let suggestion =
+        "Please style your TypeScript code using: `npx prettier --quote-props=consistent --write ./**/*.ts`"
+
+    GitRestore()
+
+    let exitCode =
+        if ContainsFiles rootDir "*.ts" then
+            InstallPrettier(prettierVersion)
+            StyleTypeScriptFiles()
+            let processResult = GitDiff()
+            PrintProcessResult processResult suggestion
+            GetProcessExitCode processResult
+
+        else
+            0
+
+    exitCode
+
+let CheckStyleOfYmlFiles(rootDir: DirectoryInfo) : int =
+    let suggestion =
+        "Please style your YML code using: `npx prettier --quote-props=consistent --write ./**/*.yml`"
+
+    GitRestore()
+
+    let exitCode =
+        if ContainsFiles rootDir "*.yml" then
+            InstallPrettier(prettierVersion)
+            StyleYmlFiles()
+            let processResult = GitDiff()
+            PrintProcessResult processResult suggestion
+            GetProcessExitCode processResult
+        else
+            0
```

Comment:
@realmarv hardcoding 0 here is a bit weird, why dont call the var `success` instead of `exitCode`, and then here you hardcode `true`

Reply:
Done